lib/puppetfactory.rb

```diff
@@ -85,6 +85,14 @@ def settings.puppetfactory
 
   # set the currently active user. This should probably be a PUT action.
   get '/users/active/:username' do |username|
+    username ||= myParams[user]
+
+    @users ||= load_users()
+
+    unless @users.include?(username)
+      return {"status" => "failure"}.to_json
```

Comment:
Add something like `"message" => "Invalid username"` too. Also, what happens on the UI? Nothing?